ramanujan/matrix_test.py

```diff
@@ -23,20 +23,20 @@ def test_gcd_reduce():
 
 def test_walk_0():
     m = Matrix([[x, 3 * x + 5 * y], [y**7 + x - 3, x**5]])
-    assert m.walk({x: 0, y: 1}, 0, {x: x, y: y}) == Matrix.eye(2)
+    assert m.walk({x: 0, y: 1}, 0, {x: x, y: y})[-1] == Matrix.eye(2)
 
 
 def test_walk_1():
```

Comment:
Lets merge the `test_walk_0` and `test_walk_1` together to a `test_walk_list_iterations`. That way we also check the list functionality.

Reply:
Added a different test instead